gpu/containers/src/initialization.cpp

```diff
@@ -1,246 +1,358 @@
+  };
+
+  std::array<SMtoCores, 15> gpuArchCoresPerSM = {{{0x30, 192},
+                                                  {0x32, 192},
+                                                  {0x35, 192},
+                                                  {0x37, 192},
+                                                  {0x50, 128},
+                                                  {0x52, 128},
+                                                  {0x53, 128},
+                                                  {0x60, 64},
+                                                  {0x61, 128},
+                                                  {0x62, 128},
+                                                  {0x70, 64},
+                                                  {0x72, 64},
+                                                  {0x75, 64},
+                                                  {0x80, 64},
+                                                  {0x86, 128}}};
+  for (const auto& sm2cores : gpuArchCoresPerSM) {
+    if (sm2cores.SM == ((major << 4) + minor))
+      return sm2cores.Cores;
+  }
+  printf(
+      "\nCan't determine number of cores. Unknown SM version %d.%d!\n", major, minor);
+  return 0;
 }
+} // namespace
 
-void pcl::gpu::printShortCudaDeviceInfo(int device)
+void
+pcl::gpu::printCudaDeviceInfo(int device)
 {
-    int count = getCudaEnabledDeviceCount();
-    bool valid = (device >= 0) && (device < count);
-
-    int beg = valid ? device   : 0;
-    int end = valid ? device+1 : count;
-
-    int driverVersion = 0, runtimeVersion = 0;
-    cudaSafeCall( cudaDriverGetVersion(&driverVersion) );
-    cudaSafeCall( cudaRuntimeGetVersion(&runtimeVersion) );
-
-    for(int dev = beg; dev < end; ++dev)
-    {                
-        cudaDeviceProp prop;
-        cudaSafeCall( cudaGetDeviceProperties(&prop, dev) );
-
-        const char *arch_str = prop.major < 2 ? " (pre-Fermi)" : "";
-        printf("[pcl::gpu::printShortCudaDeviceInfo] : Device %d:  \"%s\"  %.0fMb", dev, prop.name, (float)prop.totalGlobalMem/1048576.0f);
-        printf(", sm_%d%d%s, %d cores", prop.major, prop.minor, arch_str, convertSMVer2Cores(prop.major, prop.minor) * prop.multiProcessorCount);                
-        printf(", Driver/Runtime ver.%d.%d/%d.%d\n", driverVersion/1000, driverVersion%100, runtimeVersion/1000, runtimeVersion%100);
-    }
-    fflush(stdout);
+  int count = getCudaEnabledDeviceCount();
+  bool valid = (device >= 0) && (device < count);
+
+  int beg = valid ? device : 0;
+  int end = valid ? device + 1 : count;
+
+  printf(
+      "*** CUDA Device Query (Runtime API) version (CUDART static linking) *** \n\n");
+  printf("Device count: %d\n", count);
+
+  int driverVersion = 0, runtimeVersion = 0;
+  cudaSafeCall(cudaDriverGetVersion(&driverVersion));
+  cudaSafeCall(cudaRuntimeGetVersion(&runtimeVersion));
+
+  const char* computeMode[] = {
+      "Default (multiple host threads can use ::cudaSetDevice() with device "
+      "simultaneously)",
```

Comment:
This can be improved

Reply:
Do you mean that we can use a `std::array` again? Or do you refer to the line-breaks and lengthy text or something else altogether?

Reply:
The line breaks can be made so that we don't have sub optimal breaks

Reply:
Ok, I have changed the wording a bit to reign in these pesky linebreaks!